docs/HARNESSES.md

```diff
@@ -0,0 +1,142 @@
+## [🏠][home] &nbsp; → &nbsp; [Documentation][api] &nbsp; → &nbsp; **Component Test Harnesses**
+
+<details>
+  <summary>🧭 &nbsp;<b>Related topics</b></summary>
+
+> ### First Steps
+>
+> You may want to visit our [first steps page][firststeps].
+>
+> ### Writing Good Tests
+>
+> Introduction on [how to write simple, maintainable tests using ngtx][goodtests] and its features.
+>
+> ### Declarative Testing API
+>
+> Formal documentation of ngtx' [declarative testing api][declarativetests] helping you to write beautiful, simple and robust tests.
+
+---
+
+</details>
+
+&nbsp;
+
+This article explains what component test harnesses are and why they are useful for everyday testing.
+
+> ### Pro Tip!
+>
+> To make the most of it, combine them with ngtx' [declarative testing api][goodtests]!
+
+## Use Component Test-APIs (aka "Component Harnesses")
+
+Although being "DRY" is something absolutely normal in production code, this is often not true in testing code. But fact is, the same reasons behind us writing DRY code in production, should urge us to maintaining DRY in testing code as well.
+
+ngtx is featuring a concept that is also known as `Component Harnesses`. Harnesses are an abstraction of the component-under-test against the test suite that actually performs the tests. This way tests use a well-defined API rather than search the component's template for semantic stuff that might change in future.
+
+When the component evolves, the tests can stay the same and only the component harness needs to be adapted to the component's change. Moreover, this approach makes your code automatically DRY, since each test-case uses a common API rather than reformulating its search-queries for semantic parts of your component under test.
+
+### Example
+
+Say we have an table component listing some persons and providing basic actions:
+
+```ts
+@Component({
+  template: `
+    <table>
+      <tr>
+        <th>Name</th>
+        <th>Age</th>
+        <th>Actions</th>
+      </tr>
+      <tr *ngFor="let person of persons" data-ngtx="item-row">
+        <td>{{ person.name }}</td>
+        <td>{{ person.age }}</td>
+        <td>
+          <button>Edit</button>
+          <button>Delete</button>
+        </td>
+      </tr>
+    </table>
+  `,
+})
+class MyTableComponent {
+  @Input() persons: Person[] = [];
+}
+```
+
+When testing this component, the simplest approach is to let the test cases search for their own:
```

Comment:
I don´t have a concrete suggestion, but it would be cool to reduce the text to have a small "DO´s and DONT´s" part which shows the divs between the recommended code and the not recommended

Reply:
I googled a bit but this seems not to be possible. I'm thinking of creating an own documentation angular app for ngtx. there I would like to implement something like that. But for now we need to stick with the current way unfortunately.